src/screens/KeyBasedAssetTransfer/KeyBasedAssetTransferConfirm.js

```diff
@@ -17,192 +17,227 @@
 import Button from 'components/Button';
+import ContainerWithHeader from 'components/Layout/ContainerWithHeader';
+import FeeTable from 'components/modern/FeeTable';
+import Image from 'components/Image';
 import Spinner from 'components/Spinner';
-import { BaseText, MediumText } from 'components/Typography';
-import FeeLabelToggle from 'components/FeeLabelToggle';
-
-// utils
-import { fontStyles, spacing } from 'utils/variables';
-import { formatFullAmount, humanizeHexString } from 'utils/common';
-import { getBalance } from 'utils/assets';
+import Text from 'components/modern/Text';
 
-// constants
+// Constants
 import { ETH, COLLECTIBLES } from 'constants/assetsConstants';
 import { KEY_BASED_ASSET_TRANSFER_UNLOCK } from 'constants/navigationConstants';
 
-// selectors
-import { activeAccountAddressSelector } from 'selectors';
+// Selectors
+import { useRootSelector, useRates, useFiatCurrency, activeAccountAddressSelector } from 'selectors';
 
-// types
-import type { Balances, KeyBasedAssetTransfer } from 'models/Asset';
-import type { RootReducerState } from 'reducers/rootReducer';
+// Utils
+import { getBalanceBN, getBalanceInFiat, getFormattedBalanceInFiat } from 'utils/assets';
+import { BigNumber, formatTokenAmount, humanizeHexString } from 'utils/common';
+import { useThemedImages } from 'utils/images';
+import { spacing } from 'utils/variables';
 
+// Types
+import type { Balances, Rates, KeyBasedAssetTransfer } from 'models/Asset';
 
-type Props = {
-  navigation: NavigationScreenProp<*>,
-  keyBasedAssetsToTransfer: KeyBasedAssetTransfer[],
-  isCalculatingGas: boolean,
-  availableBalances: Balances,
-  activeAccountAddress: string,
-  keyBasedWalletAddress: ?string,
-};
-
-const DetailsTitle = styled(BaseText)`
-  ${fontStyles.regular};
-  color: #999999;
-`;
+const KeyBasedAssetTransferConfirm = () => {
+  const { t, tRoot } = useTranslationWithPrefix('smartWalletContent.confirm');
+  const navigation = useNavigation();
 
-const DetailsValue = styled(MediumText)`
-  ${fontStyles.big};
-`;
+  const rates = useRates();
+  const fiatCurrency = useFiatCurrency();
 
-const DetailsLine = styled.View`
-  padding-bottom: ${spacing.rhythm}px;
-`;
+  const assetTransfers = useRootSelector((root) => root.keyBasedAssetTransfer.data);
+  const keyWalletBalances = useRootSelector((root) => root.keyBasedAssetTransfer.availableBalances);
+  const isCalculatingGas = useRootSelector(root => root.keyBasedAssetTransfer.isCalculatingGas);
 
-const DetailsWrapper = styled.View`
-  padding: 30px ${spacing.large}px 0px ${spacing.large}px;
-`;
+  const activeAccountAddress = useRootSelector(activeAccountAddressSelector);
+  const keyBasedWalletAddress = useRootSelector(root => root.wallet.data?.address);
 
-const FooterInner = styled.View`
-  align-items: center;
-  width: 100%;
-`;
+  const images = useThemedImages();
 
-const NotEnoughFee = styled(BaseText)`
-  margin-top: ${spacing.large}px;
-`;
+  if (isCalculatingGas) {
```

Comment:
I think this answered my previous question about gas calculation...

Reply:
This seem to run the transaction in the simulation mode to get the gas price though Infura/Provider, hence await + UI feedback. BTW the transfer logic did not change, and previous version of the screen also used spinners.